tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py

```diff
@@ -89,13 +91,21 @@ def _BigQueryToElwc(pipeline: beam.Pipeline, exec_properties: Dict[str, Any],
   Raises:
     RuntimeError: Context features must be included in the queried result.
   """
+  # Try to parse the GCP project ID from the beam pipeline options.
+  beam_pipeline_args = exec_properties['_beam_pipeline_args']
+  pipeline_options = beam.options.pipeline_options.PipelineOptions(
+      beam_pipeline_args)
+  project = pipeline_options.view_as(
+      beam.options.pipeline_options.GoogleCloudOptions).project
+  if isinstance(project, value_provider.ValueProvider):
+    project = project.get()
```

Comment:
Is this duplicated from BigQueryExampleGen? if so, should we call into a common utility for this (as opposed to duplicating the code)?

Reply:
It is indeed duplicate. I can try to refactor into its own library separately. Mostly didn't want to create a separate library as part of this.

What would be a good place? tfx/extensions/google_cloud_big_query/component_utils.py?

Reply:
you can use this tfx/extensions/google_cloud_big_query/utils.py